src/selfie_park/src/Search_server.cpp

```diff
@@ -0,0 +1,288 @@
+  search_server_.start();
+  pnh_.param<float>("point_min_x", point_min_x, 0.01);
+  pnh_.param<float>("point_max_x", point_max_x, 2);
+  pnh_.param<float>("point_min_y", point_min_y, -1);
+  pnh_.param<float>("point_max_y", point_max_y, 0.2);
+  pnh_.param<float>("default_speed_in_parking_zone",
+                    default_speed_in_parking_zone, 0.3);
+  pnh_.param<bool>("visualization_in_searching", visualization, false);
+
+  speed_current.data = default_speed_in_parking_zone;
+}
+
+Search_server::~Search_server() {}
+
+bool Search_server::init() {
+  this->obstacles_sub =
+      nh_.subscribe("/obstacles", 1, &Search_server::manager, this);
+  this->visualize_lines_pub =
+      nh_.advertise<visualization_msgs::Marker>("/visualization_lines", 1);
+  this->visualize_free_place =
+      nh_.advertise<visualization_msgs::Marker>("/free_place", 1);
+  this->point_pub = nh_.advertise<visualization_msgs::Marker>("/box_points", 5);
+  this->speed_publisher = nh_.advertise<std_msgs::Float64>("/max_speed", 0.5);
+
+  speed_publisher.publish(speed_current);
+  min_spot_lenght = search_server_.acceptNewGoal()->min_spot_lenght;
+  publishFeedback(START_SEARCHING_PLACE);
+  ROS_INFO("Initialized");
+}
+
+void Search_server::manager(const selfie_msgs::PolygonArray &msg) {
+  // to save cpu time just do nothing when new scan comes
+  if (!search_server_.isActive()) {
+    ROS_INFO("search_server_ is not active");
+    return;
+  }
+  filter_boxes(msg);
+  if (visualization)
+    display_places(boxes_on_the_right_side, "FilteredBoxes");
+  // ROS_INFO("Size of  boxes_on_the_right %lu",boxes_on_the_right_side.size());
+
+  switch (action_status.action_status) {
+  case START_SEARCHING_PLACE:
+    if (find_free_places()) {
+      publishFeedback(FOUND_PLACE_MEASURING);
+      speed_current.data = 0.5;
+      speed_publisher.publish(speed_current);
+      ros::Duration(0.4).sleep();//waiting, because next measure should be taken when car is stopped
+    }
+    break;
+
+  case FOUND_PLACE_MEASURING:
+    if (find_free_places()) {
+      if (first_free_place.bottom_left.x <= 0.3)
+        publishFeedback(FIND_PROPER_PLACE);
+    } else 
+    {
+      speed_current.data = default_speed_in_parking_zone;
+      publishFeedback(START_SEARCHING_PLACE);
+    }
+    speed_publisher.publish(speed_current);
+    break;
+
+  case FIND_PROPER_PLACE:
+    if (find_free_places()) {
+      std::cout << "Found proper place\nsending result";
+      speed_current.data = 0.5;
+      send_goal();
+    } else {
+      std::cout << "Place lost\n";
+      speed_current.data = default_speed_in_parking_zone;
+    }
+    speed_publisher.publish(speed_current);
+    break;
+
+  default:
+    ROS_INFO("Err, wrong action_status");
+    break;
+  }
+}
+
+void Search_server::filter_boxes(const selfie_msgs::PolygonArray &msg) {
+  this->boxes_on_the_right_side.clear();
+  for (int box_nr = msg.polygons.size() - 1; box_nr >= 0; box_nr--) {
+    float min_x = point_min_x;
```

Comment:
Po co te przypisania? Można dać zmienne `point_min_x` po prostu do argumentów funkcji.

Reply:
Racja, to jakieś pozostałości po starym kodzie.